BL-5228 Allow user to choose background color for thumbnail

JohnThomson opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/bloombooks/bloomdesktop/1963)
<!-- Reviewable:end -->

hatton commented:
Added some comments on the youtrack card

hatton commented:
sorry, haven't reviewed code yet. Added two l10n/UI-review comments.

---

Review status: 0 of 14 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[DistFiles/localization/en/Bloom.xlf, line 2102 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1963#-KzAtYPxQ1y2iV6ZIDka:-KzAtYPxQ1y2iV6ZIDkb:b81y4pl) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e76b17c55c2cd4ebab34971537db9fb949d8ffc/DistFiles/localization/en/Bloom.xlf#L2102)):*
> ```XML
>         <source xml:lang="en">Media</source>
>         <note>ID: PublishTab.Android.Media</note>
>         <note>The top heading in the publish to Android tab. Thumbnail color and what audio and fonts to include.</note>
> ```

This note seems premature, and locks to a location unnecessarily. How about "a heading in the Publish to Android screen".

---

*[src/BloomBrowserUI/publish/android/androidPublishUI.less, line 91 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1963#-KzAsoN_hIDjwc6e2hs2:-KzAsoN_hIDjwc6e2hs3:b-juj7l9) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e76b17c55c2cd4ebab34971537db9fb949d8ffc/src/BloomBrowserUI/publish/android/androidPublishUI.less#L91)):*
> ```Less
> @tc-color-margin: 4px;
> @tc-hex-padding: 3px;
> @tc-image-margin: 10px;
> ```

Our tension here is of course, the larger it is, the smaller the thumbnail, even when we don't need any border at all.  I played around... let's go with 2px border.

I'm not seeing why you don't literally us a border, but I assume there's a good reason.

---

*[src/BloomBrowserUI/publish/android/androidPublishUI.less, line 105 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1963#-KzAuA-riIzN72922b8v:-KzAuA-riIzN72922b8w:bqaieno) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e76b17c55c2cd4ebab34971537db9fb949d8ffc/src/BloomBrowserUI/publish/android/androidPublishUI.less#L105)):*
> ```Less
> }
> 
> div.tc-image-wrapper {
> ```

Although you'll see it throughout bloom's code because 7 years ago I was just learning css, it's not normally good form to include the "div". https://cssguidelin.es/#css-selectors

---

*[src/BloomBrowserUI/publish/android/androidPublishUI.tsx, line 96 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1963#-KzBPiQ80HafniISTFFn:-KzBPiQ80HafniISTFFo:b-ttbirk) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e76b17c55c2cd4ebab34971537db9fb949d8ffc/src/BloomBrowserUI/publish/android/androidPublishUI.tsx#L96)):*
> ```XML
>     render() {
>         let self = this;
>         let colors: string[] = ["#E48C84", "#B0DEE4", "#98D0B9", "#C2B6AF", "#FF6900", "#FCB900", "#7BDCB5", "#00D084", "#8ED1FC", "#0693E3", "#ABB8C3", "#EB144C", "#F78DA7", "#9900EF"];
> ```

The 4th item has a typo. Should be C2A6BF.  Here's a list you can use while we're waiting for graphic artist feedback: "#E48C84", "#B0DEE4", "#98D0B9", "#C2A6BF", "#FFFFA4", "#FEBF00", "#7BDCB5", "#B2CC7D", "#F8B576", "#D29FEF", "#ABB8C3", "#C1EF93", "#FFD4D4", "#FFAAD4"

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1963#-:-KzBVhiQpNDC4fe4bhRi:b-7iwg5y)*
<!-- Sent from Reviewable.io -->

JohnThomson commented:
> heading in the Publish to Android screen
Fixed. See question in one of my responses.

---

Review status: 0 of 14 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[DistFiles/localization/en/Bloom.xlf, line 2102 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1963#-KzAtYPxQ1y2iV6ZIDka:-KzPLnHz9o_VPRTJKGjh:b-896fix) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e76b17c55c2cd4ebab34971537db9fb949d8ffc/DistFiles/localization/en/Bloom.xlf#L2102)):*
<details><summary><i>Previously, hatton (John Hatton) wrote…</i></summary><blockquote>

This note seems premature, and locks to a location unnecessarily. How about "a heading in the Publish to Android screen".
</blockquote></details>

Done.

---

*[src/BloomBrowserUI/publish/android/androidPublishUI.less, line 91 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1963#-KzAsoN_hIDjwc6e2hs2:-KzPM4570Dmwt6Q19ScU:bx3dsjc) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e76b17c55c2cd4ebab34971537db9fb949d8ffc/src/BloomBrowserUI/publish/android/androidPublishUI.less#L91)):*
<details><summary><i>Previously, hatton (John Hatton) wrote…</i></summary><blockquote>

Our tension here is of course, the larger it is, the smaller the thumbnail, even when we don't need any border at all.  I played around... let's go with 2px border.

I'm not seeing why you don't literally us a border, but I assume there's a good reason.
</blockquote></details>

When it's a line drawing, we need the whole background to be this color. I didn't see any point in making a distinct border the same color as the background. (I haven't tried to distinguish pictures that do and don't need the extra space...a little solid border feels like a good thing for line drawings too.)

Changed to 2px. What about the size (perhaps unfortunately a different setting) used for the border in the actual generated image? (Currently 10% of the smallest dimension, see runtimeImageProcessor line 202)

---

*[src/BloomBrowserUI/publish/android/androidPublishUI.less, line 105 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1963#-KzAuA-riIzN72922b8v:-KzPMpCpkfygcG3rtMX2:b-896fix) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e76b17c55c2cd4ebab34971537db9fb949d8ffc/src/BloomBrowserUI/publish/android/androidPublishUI.less#L105)):*
<details><summary><i>Previously, hatton (John Hatton) wrote…</i></summary><blockquote>

Although you'll see it throughout bloom's code because 7 years ago I was just learning css, it's not normally good form to include the "div". https://cssguidelin.es/#css-selectors
</blockquote></details>

Done.

---

*[src/BloomBrowserUI/publish/android/androidPublishUI.tsx, line 96 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1963#-KzBPiQ80HafniISTFFn:-KzPN-t_3rk5b0oNUVBP:b-896fix) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/0e76b17c55c2cd4ebab34971537db9fb949d8ffc/src/BloomBrowserUI/publish/android/androidPublishUI.tsx#L96)):*
<details><summary><i>Previously, hatton (John Hatton) wrote…</i></summary><blockquote>

The 4th item has a typo. Should be C2A6BF.  Here's a list you can use while we're waiting for graphic artist feedback: "#E48C84", "#B0DEE4", "#98D0B9", "#C2A6BF", "#FFFFA4", "#FEBF00", "#7BDCB5", "#B2CC7D", "#F8B576", "#D29FEF", "#ABB8C3", "#C1EF93", "#FFD4D4", "#FFAAD4"
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1963#-:-KzPOHb50azsU-e3DUAN:bowdvw7)*
<!-- Sent from Reviewable.io -->
